### PR TITLE
Fix BLEU/PPL calculation during training

### PR DESCRIPTION
**Pull request by hirofumi0810**
- Fixed PPL calculation
- Fix BLEU calculation
- Refine ST/MT scripts to keep consistency

### CODE REVIEW

File: espnet/nets/pytorch_backend/e2e_mt_transformer.py [hunk 1]
```diff
@@ -461,8 +446,6 @@ def translate(self, x, trans_args, char_list=None, rnnlm=None, use_jit=False):
                     remained_hyps.append(hyp)
 
             # end detection
-            from espnet.nets.e2e_asr_common import end_detect
-
             if end_detect(ended_hyps, i) and trans_args.maxlenratio == 0.0:
```
Comment (sw005320) [inline espnet/nets/pytorch_backend/e2e_mt_transformer.py:449]: This is not related to this PR.

I'm just curious whether this `end_detect` is working well for MT.
The [hyper-parameters](https://github.com/espnet/espnet/blob/6f6452a3891d11b0b0306084dd46a464ddf8e25c/espnet/nets/e2e_asr_common.py#L19) in the `end_detect` function were determined by ASR.
I think they are based on a similar problem and we could use the same hyper-parameters though.
@takaaki-hori, what do you think?

**Comment by hirofumi0810** [comment]
@sw005320 I pushed a tiny modification. Now it's good to me, too.
Regarding the `end_detect` function, I just moved it to the top. I can remove it if needed.

**Comment by hirofumi0810** [comment]
Maybe this is the last change.

**Comment by hirofumi0810** [comment]
bleu
ppl
BLEU and PPL are falling within reasonable range but BLUE is relatively high. 
This is probably because corpus-level BLEU scores are calculated sentence by sentence and chainer reporter summarizes the results.
To calculate the exact BLEU scores, we need to get the entire sentences and then pass them though nltk.
In order to do that, we need to modify the extension function in chainer.

**Comment by sw005320** [comment]
I think this is acceptable.
More or less, CER/accuracy has the same issue.

**Comment by sw005320** [comment]
I checked the rest of the modifications, and it looks good for me.
Many thanks!

**Comment by sw005320** [comment]
One note. This includes some refactoring, but this is still a minor modification for me. So, I merged it into the mater instead of the dev branch.
